src/pcap/cmd/pcap-bosh-cli/main.go

```diff
@@ -382,8 +390,24 @@ func (e *Environment) UpdateTokens() error {
 // connect uses the URL from the parsed Bosh environment of a config, sets up the http client for use with either TLS or plain HTTP
 // and uses this client to establish a connection to the BOSH Director and its UAA.
 func (e *Environment) connect() error {
+	err := e.setup()
+
+	if err != nil {
+		return err
+	}
+
+	err = e.fetchUAAURL()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// setup configures the BOSH http client for a given environment.
```

Comment:
in the godoc please explain the URL autosensing, especially with the new functionality of determining HTTP vs. HTTPS based on the existence of a CA certificate.

Reply:
we don't. if the URL contains no scheme, we assume HTTPS as with the pcap API. no other change was made, because there is no way to talk to BOSH director without TLS anyway.